packages/salesforcedx-vscode-apex/src/index.ts

```diff
@@ -12,23 +12,84 @@ import {
   DEBUGGER_LINE_BREAKPOINTS
 } from './constants';
 import * as languageServer from './languageServer';
+import { ApexTestOutlineProvider } from './views/testOutline';
+
+export type ApexTestRequestInfo = {
+  methodName: string;
+  parent: string;
+  line: number;
+  file: string;
+};
 
 let languageClient: LanguageClient | undefined;
 let languageClientReady = false;
 
 export async function activate(context: vscode.ExtensionContext) {
   languageClient = await languageServer.createLanguageServer(context);
-  const handle = languageClient.start();
-  context.subscriptions.push(handle);
+  if (languageClient) {
+    const handle = languageClient.start();
+    context.subscriptions.push(handle);
+
+    languageClient.onReady().then(() => {
+      languageClientReady = true;
+    });
+  }
+
+  // Test View
+  const rootPath = vscode.workspace.rootPath || '';
+  let apexClasses: ApexTestRequestInfo[] | null = null;
+  if (isLanguageClientReady()) {
+    apexClasses = (await getApexTests()) as ApexTestRequestInfo[];
+  }
+
+  const apexClassesDocs = await vscode.workspace.findFiles('**/*.cls');
+  const testOutlineProvider = new ApexTestOutlineProvider(
+    rootPath,
+    apexClassesDocs,
+    apexClasses
+  );
+
+  const testViewItems = new Array<vscode.Disposable>();
```

Comment:
Could you refactor the following parts about running test into its own method to make it cleaner? The `activate` method is getting too long.

Reply:
Look at how commands are registered on activate for salesforcedx-vscode-apex-replay-debugger and salesforcedx-vscode-core extensions.